libsplinter/src/store/command/executor.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018-2022 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[cfg(feature = "postgres")]
+use diesel::pg::PgConnection;
+use diesel::r2d2::{ConnectionManager, Pool};
+#[cfg(feature = "sqlite")]
+use diesel::sqlite::SqliteConnection;
+
+use crate::error::InternalError;
+use crate::store::command::StoreCommand;
+
+/// Provides an API for executing `StoreCommand`s
+pub trait StoreCommandExecutor {
+    type Context;
+    type Error;
```

Comment:
Is there a use case for Error being a generic type? Within the impl below, it is hard-coded to InternalError.

Reply:
Couldn't come up with a use case, removed the associated type `Error` from this trait